hoomd/hpmc/pytest/test_nec_sphere_eos.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2009-2021 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+"""Test hoomd.hpmc.integrate_nec.Sphere."""
+
+import hoomd
+import hoomd.hpmc.integrate_nec
+import hoomd.hpmc.tune
+from hoomd.conftest import operation_pickling_check
+
+import pytest
+import math
+
+# from test_sphere_eos.py for hoomd-2.9
+# see for example Guang-Wen Wu and Richard J. Sadus, doi:10.1002.aic10233
+_phi_p_ref = [(0.29054, 0.1), (0.91912, 0.2), (2.2768, 0.3), (5.29102, 0.4),
+              (8.06553, 0.45), (9.98979, 0.475)]
+rel_err_cs = 0.0015
+
+
+@pytest.mark.parametrize("betap,phi", _phi_p_ref)
+@pytest.mark.validate
+def test_sphere_eos_nec(betap, phi, simulation_factory,
```

Comment:
The validation test looks good, and is nice to have. We should also test the proper validation of getting and setting parameters and type parameters.  We also need to test the new NEC tuner class.

Reply:
I fixed the pickling test to apply tests to the NEC integrators (they were testing the normal HPMC integrators). This indirectly tests parameter get and set. Do you have time to implement a test of the tuner? I don't.